fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingCommon.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * OnlineIndexerCommon.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.metadata.RecordType;
+import com.apple.foundationdb.record.provider.foundationdb.synchronizedsession.SynchronizedSessionRunner;
+import com.apple.foundationdb.record.query.plan.synthetic.SyntheticRecordFromStoredRecordPlan;
+import com.apple.foundationdb.record.query.plan.synthetic.SyntheticRecordPlanner;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+/**
+ * Shared structure to be used (only) by the OnlineIndexer* modules.
+ */
+@API(API.Status.INTERNAL)
+public class IndexingCommon {
```

Comment:
minor: `IndexingState` might be a better name, to encapsulate that this is to be used to encapsulate shared state that is used by the indexers

Reply:
Wouldn't it be confused with `enum IndexState`?

Reply:
I would hope `IndexingState` vs `IndexState` would be easy enough to distinguish, but perhaps you're right...

I feel more confident in `State` as a suffix than `Common` (and as an abstraction), which I don't think we really use, but this shouldn't block merging the PR, I suppose.

Reply:
I agree `IndexingState` will be a good name. But whatever name to be used, the JavaDoc could explain the intention of this object in more depth.